core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -425,10 +425,9 @@ private[spark] class TaskSchedulerImpl(
           }
         }
       } else {
-        // No task sets are active but we still got an error. Just exit since this
-        // must mean the error is during registration.
-        // It might be good to do something smarter here in the future.
-        throw new SparkException(s"Exiting due to error from cluster scheduler: $message")
+        // No task sets are active but we still got an error. Just log the error
+        // since this must mean the error is during registration.
+        logError(s"Exiting due to error from cluster scheduler: $message")
```

Comment:
sorry, ignore the first comment, just took finally as catch

error() is eventually called in the rpcEndPoint of AppClient, when an exception is thrown, the endpoint (with akka impl) will restart....

so, I think we can add a `catch` before finally clause when calling `error()`